docs/ml_profiles.md

```diff
@@ -0,0 +1,101 @@
+
+---
+
+## Detailed Descriptions of Profiles
+
+### 1. **`ml-tiny`**
+- **Purpose**: Focused on minimal scans, providing only essential information for lightweight use cases.
+- **Settings**:
+  - **Deep Analysis**: Disabled.
+  - **Evidence Collection**: Disabled.
+  - **Cryptographic Dependency Inclusion**: Disabled.
+  - **Install Dependencies**: Disabled.
+- **Post-Generation Behavior**:
+  - Applies `requiresContextTrimming` to minimize the BOM by removing:
+    - `authors`
+    - `supplier`
+    - `publisher`
+    - `dependencies`
+    - `externalReferences`
+    - Other detailed fields.
+  - If generating a SaaSBOM, component details are removed, and service names are anonymized (e.g., `service-0`, `service-1`).
+- **Use Case**: Ideal for quick scans in CI/CD pipelines or scenarios where minimal information is sufficient.
+
+---
+
+### 2. **`machine-learning` / `ml`**
+- **Purpose**: The default profile for most ML projects, offering a balance between performance and depth.
+- **Settings**:
+  - **Deep Analysis**: Enabled.
+  - **Evidence Collection**: Disabled.
+  - **Cryptographic Dependency Inclusion**: Disabled.
+  - **Install Dependencies**: Enabled.
+- **Post-Generation Behavior**: No additional trimming or tuning applied.
+- **Use Case**: Suitable for general-purpose ML projects requiring a deeper analysis without unnecessary details.
+
+---
+
+### 3. **`deep-learning` / `ml-deep`**
+- **Purpose**: Designed for in-depth analysis of complex ML projects, capturing detailed evidence and cryptographic dependencies.
+- **Settings**:
+  - **Deep Analysis**: Enabled.
+  - **Evidence Collection**: Enabled.
+  - **Cryptographic Dependency Inclusion**: Enabled.
+  - **Install Dependencies**: Enabled.
+- **Post-Generation Behavior**: No trimming or tuning applied.
+- **Use Case**: Best for detailed audits or when cryptographic and evidence data are required.
+
+---
+
+## Post-Generation Behaviors
+
+Depending on the ML profile, `cdxgen` applies the following behaviors after generating the Bill of Materials (BOM):
+
+### 1. **`requiresContextTuning`**
+- Modifies the BOM to streamline the output:
+  - Removes fields such as `description`, `properties`, and `evidence`.
+  - Updates the BOM reference format using `bomLinkPrefix`.
+- **Applicable Profiles**: Not directly tied to any of the ML profiles but can be extended as needed.
+
+### 2. **`requiresContextTrimming`**
+- Minimizes the BOM to include only essential information:
+  - Strips fields such as `authors`, `supplier`, `publisher`, `bom-ref`, `externalReferences`, etc.
+  - Removes the `dependencies` section for "tiny" models.
+  - Anonymizes service names for SaaSBOMs.
+- **Applicable Profiles**: Applied to the `ml-tiny` profile for lightweight scenarios.
+
+---
+
+## Example Scenarios
+
+### **Scenario 1**: Lightweight Dependency Analysis for a Quick CI Pipeline
+- **Profile**: `ml-tiny`

```

Comment:
ml-tiny is aimed at smaller models (~ 3b parameters) and bots with a smaller context window (~ 1024). `ml` is for models with 7b to 32b parameters and `ml-deep` is for >32b parameters.

Reply:
Thank you for the feedback! I'll add details about the intended use of ml-tiny and expand the description for better clarity.

Reply:
My pleasure. Apologies, it was never documented properly, so this is already a great start. Please feel free to try out the various profiles with a tool like [AnythingLLM](https://anythingllm.com) to document the behavior. To get good results with RAG use-cases, we need a good embedding like the openai-ada and a decent model like qwen2.5 7b or 32b.

Reply:
Hello @prabhu,

Thank you for your valuable feedback! I have made the requested updates.
Please let me know if there are any further changes or refinements needed. I’m happy to make additional adjustments.

Looking forward to your feedback. 😊

Best regards,
Satwik